provisioning/service/src/Authentication/SecurityHelper.cs

```diff
@@ -10,15 +10,8 @@ internal static class SecurityHelper
     {
         public static void ValidateServiceHostName(string serviceHostName, string serviceName)
         {
-            if (string.IsNullOrWhiteSpace(serviceHostName))
-            {
-                throw new ArgumentNullException(nameof(serviceHostName));
-            }
-
-            if (string.IsNullOrWhiteSpace(serviceName))
-            {
-                throw new ArgumentNullException(nameof(serviceName));
-            }
+            Argument.AssertNotNullOrWhiteSpace(serviceName, nameof(serviceName));
```

Comment:
The danger here is that this is not a public API. Which API calls this which then inherits an ArgumentException and ArgumentNullException? Will the parameter name match what parameters they have?

I'll be looking for the public API that calls this method and making sure it translates them appropriately, so they make sense to the user who called a public method.